source/mongodb-handshake/handshake.rst

```diff
@@ -224,6 +224,49 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
+--------------------------
+Speculative Authentication
+--------------------------
+
+:since: 4.4
+
+The ``isMaster`` handshake supports a new argument, ``speculativeAuthenticate``,
+provided as a BSON document. Clients specifying this argument to ``isMaster`` will
+speculatively include the first command of an authentication handshake.
+This command may be provided to the server in parallel with any standard request for
+supported authentication mechanisms (i.e. ``saslSupportedMechs``). This would permit
+clients to merge the contents of their first authentication command with their
+``isMaster`` request, and receive the first authentication reply along with the
+``isMaster`` reply.
+
+When the mechanism is ``MONGODB-X509``, ``speculativeAuthenticate`` has the same
+structure as seen in the MONGODB-X509 conversation section in the
+`Driver Authentication spec`_.
+
+When the mechanism is ``SCRAM-SHA-1`` or ``SCRAM-SHA-256``, ``speculativeAuthenticate``
+has the same fields as seen in the conversation subsection of the SCRAM-SHA-1 and
+SCRAM-SHA-256 sections in the `Driver Authentication spec`_ with an additional ``db``
+field to specify the name of the authentication database.
+
+If the ``isMaster`` command with a ``speculativeAuthenticate`` argument succeeds,
+the client should proceed with the next step of the exchange. If the ``isMaster``
+response does not include a ``speculativeAuthenticate`` reply and the ``ok`` field
+in the ``isMaster`` response is set to 1, drivers MUST authenticate using the standard
+authentication handshake.
+
+The ``speculativeAuthenticate`` reply has the same fields, except for the ``ok`` field,
+as seen in the conversation sections for MONGODB-X509, SCRAM-SHA-1 and SCRAM-SHA-256
+in the `Driver Authentication spec`_.
+
+If an authentication mechanism is not provided either via connection string or code, but
+a credential is provided, drivers MUST use the SCRAM-SHA-256 mechanism for speculative
+authentication. If no authentication mechanism and no credential are provided, drivers
```

Comment:
Can you add something about ''saslSupportedMechs" here? Maybe: ```...drivers MUST use the SCRAM-SHA-256 mechanism for speculative authentication and drivers MUST send ``saslSupportedMechs`` ```

Reply:
I have updated the paragraph with your suggestion.